gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/BuildService.groovy

```diff
@@ -135,5 +152,17 @@ class BuildService {
 
   @ResponseStatus(HttpStatus.NOT_FOUND)
   @InheritConstructors
```

Comment:
Do you need this annotation as well as the constructor definition for these exceptions? 
Also, I think these exceptions should be pulled to their own files.

Reply:
Thanks for the comments.
Oddly enough, I had to add the constructors because it wasn't compiling .  I'll will check that bit again and see if I can remove the constructors. 
I can probably pull the exceptions to their own file as well.
Cheers

Reply:
Weird! I wouldn't be surprised if you find that groovy is weird and you need both. Thanks for checking it out!